<h1>Drill: Describing Data</h1>
<h2>Question 1:</h2>
<p>Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?</p>

In [85]:
import pandas as pd
import numpy as np
import statistics as stats

names = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']
data = pd.DataFrame(index = names)


data['age'] = [14, 12, 11, 10, 8, 6, 8]

data.head(7) #Why don't you show me the chart?

print('Mean age:', np.mean(data['age']))
print('Median age: ', np.median(data['age']))
print('Mode age:', stats.mode(data['age']))

print('\n')

print('Variance: ', data['age'].var())
print('Standard Deviation: ', np.std(data['age'], ddof = 1))
print('Standard Error: ', np.std(data['age'], ddof = 1) / np.sqrt(len(data['age'])))

Mean age: 9.857142857142858
Median age:  10.0
Mode age: 8


Variance:  7.476190476190475
Standard Deviation:  2.734262327610589
Standard Error:  1.0334540197243192


<h2>Question 2:</h2>
<p>Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?</p>

<h3>Answer:</h3>
<p>I think the answer to this largely depends on what the statistic is being used for. For example, if I wanted to predict the age of the next character the producers would add to the show, I would use the mean of their age and the standard error as an inverse measure of confidence.</p>

<h2>Question 3:</h2>
<p>Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?</p>

In [86]:
data.loc['Cindy'] += 1
print('Mean age:', np.mean(data['age']))
print('Median age: ', np.median(data['age']))
print('Mode age:', stats.mode(data['age']))

print('\n')

print('Variance: ', data['age'].var())
print('Standard Deviation: ', np.std(data['age'], ddof = 1))
print('Standard Error: ', np.std(data['age'], ddof = 1) / np.sqrt(len(data['age'])))

Mean age: 10.0
Median age:  10.0
Mode age: 8


Variance:  6.333333333333333
Standard Deviation:  2.516611478423583
Standard Error:  0.9511897312113418


<h3>Answer:</h3>
<p>The only measure of central tendancy to change was the mean age of the actors. However, all measures of variance were reduced.</p>

<h2>Question 4:</h2>
<p>Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?</p>

In [87]:
data = data.drop(['Oliver']) #Correct me if I am wrong, but the .drop() method appears to return a copy of the datafram

data.loc['Jessica'] = [1]

print('Data is: ', data)

print('\n')

print('Mean age:', np.mean(data['age']))
print('Median age: ', np.median(data['age']))
print('Mode age:', 'NA')

print('\n')

print('Variance: ', data['age'].var())
print('Standard Deviation: ', np.std(data['age'], ddof = 1))
print('Standard Error: ', np.std(data['age'], ddof = 1) / np.sqrt(len(data['age'])))


Data is:           age
Greg      14
Marcia    12
Peter     11
Jan       10
Bobby      8
Cindy      7
Oliver     8
Jessica    1


Mean age: 8.875
Median age:  9.0
Mode age: NA


Variance:  15.553571428571429
Standard Deviation:  3.943801646707328
Standard Error:  1.394344444020712


<h3>Answer:</h3>
<p>This renders the 'Mode' measurement unusable as a method of measurement, and with such a small dataset of discrete values, frequencyt measurements are inherently less useful.</p>

<h2>Question 5:</h2>
<p>On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?</p>

In [88]:
magNames = ['TV Guide','Entertainment Weekly','Pop Culture Today','SciPhi Phanatic']
mag_data = pd.DataFrame(index = magNames)
mag_data['Percent_Fans'] = [0.20,0.23,0.17,0.05]

mag_data.head()

print('Expected Value: ', np.mean(mag_data['Percent_Fans']))


Expected Value:  0.16250000000000003


<h3>Answer:</h3>
<p>Assuming that the tests performed by the magazines used unbiased samples of their readership and magazine readers represent an unbiased sample of the adult American population, I would estimate that 16.25% of all American adults are fans of the Brady show.</p>